source/extensions/tracers/opentelemetry/http_trace_exporter.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "envoy/config/core/v3/http_service.pb.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/http/async_client_impl.h"
+#include "source/common/http/async_client_utility.h"
+#include "source/common/http/headers.h"
+#include "source/common/http/message_impl.h"
+#include "source/common/http/utility.h"
+
+#include "opentelemetry/proto/collector/trace/v1/trace_service.pb.h"
+#include "trace_exporter.h"
+#include "tracer_stats.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+/**
+ * Exporter for OTLP traces over HTTP.
+ */
+class OpenTelemetryHttpTraceExporter : public OpenTelemetryTraceExporter,
+                                       public Http::AsyncClient::Callbacks {
+public:
+  OpenTelemetryHttpTraceExporter(Upstream::ClusterManager& cluster_manager,
+                                 envoy::config::core::v3::HttpService http_service,
+                                 OpenTelemetryTracerStats& tracing_stats);
+
+  // The default path to use when OpenTelemetryConfig::HttpConfig::traces_path is empty
+  const Http::LowerCaseString TRACES_PATH{"/v1/traces"};
+
+  bool log(const ExportTraceServiceRequest& request) override;
+
+  // Http::AsyncClient::Callbacks.
+  void onSuccess(const Http::AsyncClient::Request&, Http::ResponseMessagePtr&&) override;
+  void onFailure(const Http::AsyncClient::Request&, Http::AsyncClient::FailureReason) override;
+  void onBeforeFinalizeUpstreamSpan(Tracing::Span&, const Http::ResponseHeaderMap*) override {}
+
+private:
+  Upstream::ClusterManager& cluster_manager_;
+  envoy::config::core::v3::HttpService http_service_;
+  OpenTelemetryTracerStats& tracing_stats_;
```

Comment:
Can you add a comment about the lifetime expectation of the tracing_stats_ object? (specifically why is it assumed that it will outlive the TraceExplorer)

Reply:
I'm not sure we need this new type now.. please read my thoughts on the other comment: https://github.com/envoyproxy/envoy/pull/29207#discussion_r1342519254

Reply:
Done in: [ba5d943](https://github.com/envoyproxy/envoy/pull/29207/commits/ba5d94396faa2fd2ddf8d0d1064f97790706f2c6)